src/Host/Broker/Impl/Startup/Program.cs

```diff
@@ -57,7 +61,38 @@ public class Program {
 
             var tlsConfig = new TlsConfiguration(_logger, _securityOptions);
             var httpsOptions = tlsConfig.GetHttpsOptions(Configuration);
-            CreateWebHost(httpsOptions).Run();
+
+            Uri uri = GetServerUri(Configuration);
+            if (IsPortAvailable(uri?.Port)) {
```

Comment:
https://github.com/aspnet/Hosting/blob/2bddba8f9023c0d14332b90235c7dffe16e3e353/src/Microsoft.AspNetCore.Hosting/WebHostExtensions.cs#L65

https://github.com/aspnet/Hosting/blob/2bddba8f9023c0d14332b90235c7dffe16e3e353/src/Microsoft.AspNetCore.Hosting/Internal/WebHost.cs#L244

Looks like they're disposing the web host unconditionally when `Run` exits, and there are no hooks provided to intercept and log the exception before then. And the host, when it is disposed, disposes the service container with all its services.

So we can either rewrite `Run` (should be doable since it's just an extension method, but it's not as small as I'd like), or else just recreate the logger provider.